benchexec/tools/cpachecker.py

```diff
@@ -180,16 +180,40 @@ def option_present(option):
                         f"Unsupported data_model '{data_model}' defined for task '{task}'"
                     )
 
+        if isinstance(task.options, dict) and "witness" in task.options.keys():
+            if not option_present("witness"):
+                possible_witness_files = list(
+                    filter(
+                        lambda x: x.endswith(task.options.get("witness")),
+                        task.input_files_or_empty,
+                    )
```

Comment:
This is not really "Pythonic". In general, many developers (including the Python founder) prefer to avoid `map` and `filter` in Python in favor of comprehensions. Sometimes the former can make sense, but here it seems clear that something like the following is better:

```python
possible_witness_files = [input for input in task.input_files_or_empty if input.endswith(witness)]
```

(Same below).

However, `endswith` could lead to wrong matches in principle, couldn't it? Maybe do some path-name matching?